src/v/reflection/async_adl.h

```diff
@@ -28,6 +33,33 @@ struct async_adl {
     }
 };
 
+/// Specializations of async_adl for nested types, optional and vector
```

Comment:
Why is a specialisation for `std::optional` necessary?

Reply:
Because in the second commit of this PR this line necessitates the change https://github.com/vectorizedio/redpanda/pull/723/commits/377de4448eb585e7efae457248abe1bfa8e8ea13#diff-97ee2f298ede6e7ae833ec3b81a3a87b5e78302890883a506e82ac9bcc059a46R111.

In short optional<record_batch_reader> is a type that were now expecting over the wire from the wasm engine.

Reply:
Got it.